### PR TITLE
Move /connect/in-app-wallets to app router

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a new `in-app-wallets` feature to the dashboard, including support for API keys, a layout for the wallet interface, and a header component. It also refines navigation and user experience for managing in-app wallets.

### Detailed summary
- Added `getInAppWalletSupportedAPIKeys` function to fetch API keys.
- Created `Layout` component for the wallet interface.
- Implemented `InAppWalletsAPIKeysMenu` for API key selection.
- Developed `PageHeader` component for in-app wallets page.
- Updated `Page` to handle authentication and API key redirects.
- Enhanced `EmbeddedWallets` component to use a default tab.
- Adjusted routing and redirection logic for user navigation.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4894?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4894** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4894?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4893** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4893?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4892** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4892?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4891** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4891?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { Suspense } from "react";
+
+export default function Layout(props: { children: React.ReactNode }) {
+  return <Suspense fallback={null}>{props.children}</Suspense>;
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/layout.tsx:4]: just for my own understanding why do we need this?

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/page.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    }
+    redirect(`/login?${params.toString()}`);
+  }
+
+  const apiKeys = await getApiKeys();
+
+  return <InAppWalletsPage defaultClientId={clientId} apiKeys={apiKeys} />;
+}
+
+export const dynamic = "force-dynamic";
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/page.tsx:28]: same thing here, do we need this if we're already accessing cookies?

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/InAppWalletsPage.tsx [hunk 3]
```diff
@@ -88,50 +73,35 @@ const DashboardConnectEmbeddedWallets: ThirdwebNextPage = () => {
...
               selectedKey={selectedKey}
-              onSelect={setSelectedKey}
+              onSelect={(key) => {
+                setSelectedKey(key);
+                // add clientId search param to the url without affecting router
+                window.history.replaceState(
+                  {},
+                  "",
+                  `/dashboard/connect/in-app-wallets?clientId=${key.key}`,
+                );
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/InAppWalletsPage.tsx:83]: I wonder if this *should* affect router and we instead pass only the specific apiKey down every time? why not let the router do the filtering?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/layout.tsx (hunk 1 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/layout.tsx:4]: we don't need it anymore, let me delete that

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/page.tsx (hunk 2 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/page.tsx]: removed

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/InAppWalletsPage.tsx (hunk 3 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/InAppWalletsPage.tsx]: updated - added clientId in the route

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  searchParams: {
+    tab?: string;
+  };
+}) {
+  const authToken = getAuthToken();
+  const { clientId } = props.params;
+
+  if (!authToken) {
+    redirect(
+      `/login?next=${encodeURIComponent("/dashboard/connect/analytics")}`,
```
Comment (MananTank) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx:21]: change this to /in-app-wallets

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    redirect("/dashboard/connect/in-app-wallets");
+  }
+
+  return (
+    <div>
+      {/* header */}
+      <div className="flex flex-col gap-4 lg:flex-row lg:justify-between">
+        <PageHeader />
+        <div>
+          <InAppWalletsAPIKeysMenu apiKeys={apiKeys} selectedAPIKey={apiKey} />
```
Comment (jnsdls) [inline apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx:38]: map it so it doesn't need the full key?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx (hunk 5 above)
Comment (MananTank) [reply apps/dashboard/src/app/(dashboard)/dashboard/connect/in-app-wallets/[clientId]/page.tsx]: updated

**Review by jnsdls — APPROVED** [review]

**Pull request merged**